services/maven-metadata/maven-metadata.service.js

```diff
@@ -52,10 +53,19 @@ module.exports = class MavenMetadata extends BaseXmlService {
     })
   }
 
-  async handle(_namedParams, { metadataUrl }) {
+  async handle(_namedParams, { metadataUrl, latestOrRelease }) {
     const data = await this.fetch({ metadataUrl })
+    let v
+    if (
+      latestOrRelease === 'latest' ||
+      typeof latestOrRelease === 'undefined'
+    ) {
+      v = data.metadata.versioning.latest
+    } else if (latestOrRelease === 'release') {
+      v = data.metadata.versioning.release
+    }
     return renderVersionBadge({
-      version: data.metadata.versioning.versions.version.slice(-1)[0],
+      version: v,
     })
```

Comment:
Just FYI, a fairly common pattern in our service classes is to utilize another function `transform` that handles the concern of extracting the target elements from the raw response, and if necessary mapping or transforming that. That's not necessary here due to the simplicity, but sharing for future awareness as it's easier to test and often a bit cleaner too.

We also default our version badges to showing the latest stable/release version (`release` here), with the `include_prereleases` query param available for users that want to display the absolute latest regardless of stable/release vs. prerelease. However, even in the default case if there are no "release" versions, we will still automatically fallback to a prerelease-type version vs. throwing an error.

_If_ both the `latest` and `release` elements are always present and already reflecting that pattern, then we should be fine with a simple conditional. However, if `release` is sometimes absent then we'd need to account for that fallback pattern too